crates/static-analysis-kernel/src/analysis/javascript.rs

```diff
@@ -31,6 +86,25 @@ struct StellaExecution {
     console: Vec<String>,       // the log lines from console.log
 }
 
+/// A wrapper around a [`tree_sitter::Tree`] that can be used as a [`deno_core::Resource`].
+///
+/// This allows a Deno `op` to access a tree-sitter tree.
+#[repr(transparent)]
+pub struct ResourceTree(Arc<tree_sitter::Tree>);
+
+impl ResourceTree {
+    pub fn new(tree: Arc<tree_sitter::Tree>) -> Self {
+        Self(tree)
+    }
+
+    pub fn tree(&self) -> &tree_sitter::Tree {
+        use std::borrow::Borrow;
+        self.0.borrow()
+    }
+}
+
+impl deno_core::Resource for ResourceTree {}
```

Comment:
We should isolate this somewhere else that clearly defines where the runtime is. We may have more types like this, what about a place like `src/analysis/javascript/runtime`? The current file could then go to `src/analysis/javascript/main.rs` or something similar